tools/src/main/java/org/apache/kafka/trogdor/workload/ProduceBenchWorker.java

```diff
@@ -0,0 +1,337 @@
+        if (!running.compareAndSet(false, true)) {
+            throw new IllegalStateException();
+        }
+        log.info("{}: Activating ProduceBenchWorker.", id);
+        this.executor = Executors.newScheduledThreadPool(1,
+            ThreadUtils.createThreadFactory("ProduceBenchWorkerThread%d", false));
+        this.status = status;
+        this.doneFuture = doneFuture;
+        executor.submit(new ValidateSpec());
+    }
+
+    private static String topicIndexToName(int topicIndex) {
+        return String.format("topic%05d", topicIndex);
+    }
+
+    private void abort(String what, Exception e) throws KafkaException {
+        log.warn(what + " caught an exception: ", e);
+        doneFuture.completeExceptionally(new KafkaException(what + " caught an exception.", e));
+        throw new KafkaException(e);
+    }
+
+    public class ValidateSpec implements Callable<Void> {
+        @Override
+        public Void call() throws Exception {
+            try {
+                if (spec.activeTopics() == 0) {
+                    throw new InvalidPartitionsException("Can't have activeTopics == 0.");
+                }
+                if (spec.totalTopics() < spec.activeTopics()) {
+                    throw new InvalidPartitionsException(String.format(
+                        "activeTopics was %d, but totalTopics was only %d.  activeTopics must " +
+                            "be less than or equal to totalTopics.", spec.activeTopics(), spec.totalTopics()));
+                }
+                executor.submit(new CreateBenchmarkTopics());
+            } catch (Exception e) {
+                abort("ValidateSpec", e);
+            }
+            return null;
+        }
+    }
+
+    public class CreateBenchmarkTopics implements Callable<Void> {
+        private final static int MAX_BATCH_SIZE = 10;
+
+        @Override
+        public Void call() throws Exception {
+            try {
+                Properties props = new Properties();
+                props.put(AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, spec.bootstrapServers());
+                try (AdminClient adminClient = AdminClient.create(props)) {
+                    TreeSet<String> topicsToCreate = new TreeSet<>();
+                    for (int i = 0; i < spec.totalTopics(); i++) {
+                        topicsToCreate.add(topicIndexToName(i));
+                    }
+                    log.info("Creating " + spec.totalTopics() + " topics...");
+                    List<Future<Void>> futures = new ArrayList<>();
+                    while (!topicsToCreate.isEmpty()) {
+                        List<NewTopic> newTopics = new ArrayList<>();
+                        for (int i = 0; i < MAX_BATCH_SIZE; i++) {
+                            String topic = topicsToCreate.pollFirst();
+                            topicsToCreate.remove(topic);
+                            newTopics.add(new NewTopic(topic, 1, REPLICATION_FACTOR));
```

Comment:
This is assuming that `totalTopics` is always a multiple of `MAX_BATCH_SIZE. Is that always true? Perhaps it is better not to make that assumption in any case.

Reply:
Oh, this was not supposed to have that assumption.  Fixed.